gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -102,32 +107,33 @@
   // An AtomicReference is still used here for the compareAntSet operation.
   private final AtomicReference<ForkState> forkState;
 
+  private final GobblinMetrics forkMetrics;
+
   public Fork(TaskContext taskContext, TaskState taskState, Object schema, int branches, int index,
-      CountDownLatch countDownLatch)
-      throws Exception {
+      CountDownLatch countDownLatch) throws Exception {
 
     this.logger = LoggerFactory.getLogger(Fork.class.getName() + "-" + index);
 
     this.taskContext = taskContext;
-    this.taskState = taskState;
-    this.taskId = taskState.getTaskId();
+    this.taskStateCopy = new TaskState(taskState);
```

Comment:
This could just be `this.taskState = new TaskState(taskState)`, given that we are not also storing the original. But I don't really have a preference either way.